### PR TITLE
[ALLUXIO-1735] Using MireDot for REST API documentation

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://tachyon.atlassian.net/browse/ALLUXIO-1735

After `mvn test`, the `$ALLUXIO_HOME/core/server/target/miredot` will contain autogenerated HTML documenting the REST API. See the attached screenshots for a couple of examples.
screen shot 2016-02-25 at 3 16 02 pm
screen shot 2016-02-25 at 3 17 26 pm

@aaudiber

### CODE REVIEW

File: core/server/pom.xml [hunk 1]
```diff
@@ -162,6 +178,50 @@
...
+        <executions>
+          <execution>
+            <goals>
+              <goal>restdoc</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <license>
+            UHJvamVjdHxvcmcuYWxsdXhpby5hbGx1eGlvLWNvcmUtc2VydmVyfDIwMTctMDItMjV8dHJ1ZSNNQzBDRlFDUm1SMXpVb3dMOUkyNXZrcGtJOVlmOHY3M2tnSVVIa0NxSFgvdUlYZmd3SHdSa1hFYUFMRDQ3Q0E9
```
Comment (apc999) [inline core/server/pom.xml]: is this license particular to alluxio? if yes, probably put a comment there.

**Comment by apc999** [comment]
left one comment

File: core/server/pom.xml (hunk 1 above)
Comment (jsimsa) [inline core/server/pom.xml]: done ... fyi, I contacted the miredot guys to make sure it is okay to use their plugin in the open source project and they offered to provide us with a commercial version of the plugin for the purpose of generating the REST API ... I will update the license accordingly one I receive it from them

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**